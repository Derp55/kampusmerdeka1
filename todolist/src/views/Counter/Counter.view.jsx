import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { cuonterAction } from "../../config/redux/counter/counterSlice";

const Counter = () => {
  const state = useSelector((state) => state.counter);
  const dispatch = useDispatch();

  return (
    <div>
      <div>Counter {state.angka} </div>
      {/* {state.counter.map((val) => {
        return <div>{val}</div>;
      })} */}
      <div>
        <button
          onClick={() => {
            dispatch(cuonterAction.increment());
          }}
        >
          plus
        </button>
        <button
          onClick={() => {
            dispatch(cuonterAction.decrement());
          }}
        >
          minus
        </button>
        <button
          onClick={() => {
            dispatch(cuonterAction.reset());
          }}
        >
          reset
        </button>

        <button
          onClick={() => {
            dispatch(cuonterAction.update(1000));
          }}
        >
          update to 1000
        </button>
      </div>
    </div>
  );
};

export default Counter;
