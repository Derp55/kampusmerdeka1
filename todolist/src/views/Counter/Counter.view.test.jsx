import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../../config/redux/store";
import Counter from "./Counter.view";

describe("Counter", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Counter />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
