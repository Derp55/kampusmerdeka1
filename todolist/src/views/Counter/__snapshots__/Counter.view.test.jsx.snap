// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > should render correctly 1`] = `
<div>
  <div>
    Counter 
    110
     
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      plus
    </button>
    <button
      onClick={[Function]}
    >
      minus
    </button>
    <button
      onClick={[Function]}
    >
      reset
    </button>
    <button
      onClick={[Function]}
    >
      update to 1000
    </button>
  </div>
</div>
`;
